### ISSUE TITLE
Create of new vMWare vm crashes on plan

### ISSUE THREAD

**Issue by ntmggr** [terraform-providers/terraform-provider-vsphere#409]
### Terraform Version
```
Terraform v0.11.3
+ provider.vsphere v1.3.2
```
### vSphere Provider Version

```
terraform providers
.
├── provider.vsphere ~> 1.1
└── module.vm
    └── provider.vsphere (inherited)
```

`ls .terraform/plugins/linux_amd64/terraform-provider-vsphere_v1.3.2_x4`

### Affected Resource(s)
- `vsphere_virtual_machine`

### Terraform Configuration Files

```
variable "datacenter" {}

variable "esxi_host" {}

variable "resource_pool" {}

variable "datastore_name" {}

variable "network_name" {}

variable "template_name" {}

variable "virtual_machine_count" {}

variable "virtual_machine_cpu_count" {}

variable "virtual_machine_memory" {}

variable "virtual_machine_name_prefix" {}

variable "virtual_machine_domain" {}

variable "virtual_machine_ip_address" {}

variable "virtual_machine_netmask" {}

variable "virtual_machine_gateway" {}

variable "virtual_machine_dns_servers" {}

variable "virtual_machine_net_timeout" {
  default = false
}

data "vsphere_datacenter" "datacenter" {
  name = "${var.datacenter}"
}

data "vsphere_host" "hosts" {
  name          = "${var.esxi_host}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

data "vsphere_datastore" "datastore" {
  name          = "${var.datastore_name}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

data "vsphere_network" "network" {
  name          = "${var.network_name}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

data "vsphere_resource_pool" "resource_pool" {
  name          = "${var.resource_pool}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

data "vsphere_virtual_machine" "template" {
  name          = "${var.template_name}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

resource "vsphere_virtual_machine" "virtual_machine" {
  count                      = "${var.virtual_machine_count}"
  name                       = "${var.virtual_machine_name_prefix}"
  num_cpus                   = "${var.virtual_machine_cpu_count}"
  memory                     = "${var.virtual_machine_memory}"
  wait_for_guest_net_timeout = "${var.virtual_machine_net_timeout}"

  resource_pool_id           = "${data.vsphere_resource_pool.resource_pool.id}"
  host_system_id             = "${data.vsphere_host.hosts.id}"
  datastore_id               = "${data.vsphere_datastore.datastore.id}"
  guest_id                   = "${data.vsphere_virtual_machine.template.guest_id}"

  network_interface {
    network_id   = "${data.vsphere_network.network.id}"
    adapter_type = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }

  disk {
    label = "${var.virtual_machine_name_prefix}.vmdk"
    size = "${data.vsphere_virtual_machine.template.disks.0.size}"
  }

  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"

    customize {
      linux_options {
        host_name = "${var.virtual_machine_name_prefix}"
        domain    = "${var.virtual_machine_domain}"
      }

      network_interface {
        ipv4_address = "${var.virtual_machine_ip_address}"
        ipv4_netmask = "${var.virtual_machine_netmask}"
      }

      ipv4_gateway    = "${var.virtual_machine_gateway}"
      dns_suffix_list = ["${var.virtual_machine_domain}"]
      dns_server_list = ["${var.virtual_machine_dns_servers}"]
    }
  }
}

```

### Debug Output
Output from plan

https://gist.github.com/ntmggr/65079f6d823f7e8c62b55671f4fc63ea

### Panic Output

https://gist.github.com/ntmggr/919d97273572c1a52743b0567ebfda3b

### Expected Behavior

Should connect to vcenter and create a VM

### Actual Behavior

https://gist.github.com/ntmggr/65079f6d823f7e8c62b55671f4fc63ea

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

### Important Factoids

No

### References

N/A

**Comment by ntmggr** [comment]
I was able to track down the reason for this. The vCenter's license had expired.
I think better error handling on this would be ideal. Feel free to either close or file it as a feature request

**Comment by vancluever** [comment]
@ntmggr thanks for the extra detail! We definitely don't want a crash, but I want to make sure we detect an expired license at the correct place. I would imagine not all operations have failed under this circumstance considering you got this far.

Can you send along any other details as to what you _could_ and _could not_ do under these circumstances? This will help us in figuring out where we need to detect this and block usage of the provider.

**Comment by ntmggr** [comment]
@vancluever it was failing when it was trying to create the vm. so the data queries all worked but when it was going to a point to create the VM this is where the crash happened. The `Actual Behavior` gist shows the exact point of failure

**Comment by vancluever** [comment]
Hey @ntmggr, we have a fix in place for this now that was merged in #414. I think this will cover your scenario as well as the location of the crash was the same as #411. I'd imagine this would be hard for you to test as you have probably renewed your vSphere license already, but if you experience this problem again please open a new issue!
